script/constructron.lua

```diff
@@ -1435,7 +1435,9 @@ me.on_built_entity = function(event) -- for entity creation
                     global.ghost_tick = event.tick
                 end
             end
-        elseif entity.name == 'constructron' or entity.name == "constructron-rocket-powered" then
+        elseif entity.name == 'constructron' or entity.name == "constructron-rocket-powered"
+        or entity.name == "constructron-ss-space-spidertron" 
+        or entity.name == "constructron-spidertronmk2" or entity.name == "constructron-spidertronmk3" then

```

Comment:
Same comment here: 

``` lua
-- somewhere else
local is_valid_constructron = {
    ["constructron"] = true,
    ["constructron-rocket-powered"] = true,
    ["constructron-ss-space-spidertron"] = true,
    ["constructron-spidertronmk2"] = true,
    ["constructron-spidertronmk3"] = true,
}
```

```suggestion
        elseif is_valid_constructron[entity.name] then
```

This applies to all the bloated `if` statements.

Reply:
Or even better. You can replace the redeclaration of all spiders with this loop, after you've declared `valid_constructrons`
``` lua
-- somewhere in the main chunk, not inside an event
local is_valid_constructron = {}
for _, name in pairs(valid_constructrons) do
    is_valid_constructron[name] = true
end
```